### PR TITLE
#165637882 implement feedback

### PR DESCRIPTION
**Pull request by chuxmykel**
#### What does this PR do?
The PR implements all backend related feedback from the review session.

#### Description of the task to be completed?
- work on "failed sign in" error message
- use jwt call back methods to handle errors and remove try catch blocks
- fix debit and credit transactions on draft accounts
- handle general server errors gracefully at the app entry point.

#### How should this be manually tested?
- Clone the repo, CD into it and RUN ```npm install```
- Create a .env file and set the following values according to your needs 
```SECRET_KEY = string```
```SALT_ROUNDS = number```
```SERVER_MAIL = your_email@gmail.com```
```MAIL_PASSWORD = yourpassword```
```DATABASE_URL= your database URL```
```TEST_DATABASE_URL= your test database URL```

- Using postman, test the ALL endpoints with the following
 > Headers
  -  _key: Content-Type value: application/json_
  - _key: Content-Type value: application/x-www-form-urlencoded_

- And for protected routes, the user must be authenticated. So add the following header: *(you can get a token from ```/auth/signin``` endpoint)*
  -  _key: Authorization value: JWT -Token_

#### What are the relevant pivotal tracker stories?
- [#165637882](https://www.pivotaltracker.com/story/show/165637882)

#### Screenshots
Screenshot from 2019-04-26 19-44-30
Screenshot from 2019-04-26 19-44-05

### CODE REVIEW

File: server/v1/app/controllers/accountController.js [hunk 1]
```diff
@@ -121,7 +135,7 @@ class AccountController {
     if (response.rowCount < 1) {
       return res.status(404).json({
         status: res.statusCode,
-        error: `Account Number ${accountNumber} does not exist!`,
+        error: 'Accountdoes not exist!',
```
Comment (slimsolz) [inline server/v1/app/controllers/accountController.js]: add a space after Account

File: server/v1/app/middlewares/authenticateUser.js [hunk 2]
```diff
@@ -106,30 +104,25 @@ class AuthenticateUser {
...
-      return next();
-    } catch (error) {
-      return res.status(401).send({
+      req.user = decoded;
+    });
+    if (!req.user.isAdmin) {
+      return res.status(403).send({
         status: res.statusCode,
-        error: 'Authentication Failed',
+        error: 'You are not authorized to access this endpoint',
```
Comment (slimsolz) [inline server/v1/app/middlewares/authenticateUser.js]: change this error message to unauthorized

**Review by slimsolz — CHANGES_REQUESTED** [review]
attend to the comment

File: server/v1/app/controllers/accountController.js (hunk 1 above)
Comment (chuxmykel) [reply server/v1/app/controllers/accountController.js]: @slimsolz Done

**Review by slimsolz — APPROVED** [review]
LGTM

**Pull request merged**